source/javascripts/components/StepProperties.tsx

```diff
@@ -1,17 +1,205 @@
+
+const useVersionChange = (step: Step) => {
+	const { isOpen, onOpen, onClose } = useDisclosure();
+
+	const [{ oldHashKey, oldVersion, oldInputNames }, dispatch] = useReducer(
+		(state: ReducerState, partialState: Partial<ReducerState>) => ({ ...state, ...partialState }),
+		DefaultState,
+	);
+
+	const isSameStep = oldHashKey === step.$$hashKey;
+
+	// Calculate input changes between different step versions
+	const newInputNames = extractInputNames(step);
+	const removedInputs = oldInputNames.filter((name) => !newInputNames.includes(name));
+	const newInputs = newInputNames.filter((name) => !oldInputNames.includes(name));
+	const hasInputChanged = removedInputs.length > 0 || newInputs.length > 0;
+
+	// Calculate version changes
+	const newVersion = step.version;
+	const isVersionChanged = Boolean(oldVersion && newVersion && oldVersion !== newVersion);
+	const isMajorChange = Boolean(oldVersion && newVersion && semverService.isMajorVersionChange(oldVersion, newVersion));
+	const shouldOpenChangeDialog = isSameStep && (isMajorChange || (isVersionChanged && hasInputChanged));
+
+	useEffect(() => {
+		if (shouldOpenChangeDialog) {
+			onOpen();
+		}
+	}, [shouldOpenChangeDialog, onOpen, newVersion]);
+
+	useEffect(() => {
+		if (step.$$hashKey !== oldHashKey) {
+			dispatch(DefaultState);
+		}
+	}, [oldHashKey, step.$$hashKey]);
+
+	const result = {
+		isOpen,
+		close: onClose,
+		isMajorChange,
+		releaseNotesUrl: extractReleaseNotesUrl(step),
+		removedInputs,
+		newInputs,
+		createSnapshot: ({ oldHashKey, oldVersion, oldInputNames }: ReducerState) =>
+			dispatch({
+				oldHashKey,
+				oldVersion,
+				oldInputNames,
+			}),
+	};
+
+	return result;
+};
 
 type StepPropertiesProps = {
-  step: Step;
+	step: Step;
+	versionsWithRemarks: Array<StepVersionWithRemark>;
+	onChange: OnStepPropertyChange;
 };
 
-const StepProperties = ({ step }: StepPropertiesProps) => {
-  return (
-    <Box display="flex" flexDirection="column" p="24" gap="24">
-      <Text>Properties</Text>
-      {step.displayName()}
-    </Box>
-  );
+const StepProperties = ({ step, versionsWithRemarks, onChange }: StepPropertiesProps) => {
+	const { name, version, sourceURL, summary, description, isLibraryStep } = extractStepFields(step);
+	const [showMore, setShowMore] = useState(false);
+
+	const { register, setValue, handleSubmit } = useForm<Record<"name" | "version", string>>({
+		defaultValues: { name, version },
+	});
+	const { isOpen, close, createSnapshot, releaseNotesUrl, isMajorChange, removedInputs, newInputs } =
+		useVersionChange(step);
+	const handleChange = handleSubmit((values) => {
+		createSnapshot({
+			oldHashKey: step.$$hashKey,
+			oldVersion: step.defaultStepConfig.version,
+			oldInputNames: extractInputNames(step),
+		});
+		onChange(values);
+	});
+
+	useEffect(() => {
+		setValue("name", name);
+		setValue("version", version);
+	}, [name, version, setValue]);
```

Comment:
Investigate why is it necessary here. Maybe the useForm stores the default values and it doesn't re-initialize on step changes.

Reply:
Unfortunately, that's the only way I could get it work.